src/components/ModalWithForm.js

```diff
@@ -24,4 +27,12 @@ export default class ModalWithForm extends Modal {
       this._handleFormSubmit(this._getInputValues())
     );
   }
+
+  showButtonLoading(text) {
+    this._modalSubmitButton.textContent = text;
```

Comment:
Actually you duplicate the same code of text changing in `showButtonLoading` and `hideButtonLoading`. Please, create only one method `changeButtonText` or `renderLoading`


```js
  renderLoading(text) {
    this._modalSubmitButton.textContent = text;
  }

```